gtnet/.ipynb_checkpoints/sequence-checkpoint.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+import skbio
+
+__all__ = ['_get_DNA_map',
+           'get_sequences',
+          'chunk_seq',
+           'map_seq',
+           'get_rev_comp',
+          'get_bidir_seq']
+
+def _get_DNA_map():
+    '''
+    create a DNA map with some redundancy so that we can
+    do base-complements with +/% operations.
+    Using this scheme, the complement of a base should be:
+    (base_integer + 9) % 18
+    chars[(basemap[ord('N')]+9)%18]
+    For this to work, bases need to be ordered as they are below
+    '''
+    chars = ('ACYWSKDVN'
+             'TGRWSMHBN')
+    basemap = np.zeros(128, dtype=np.uint8)
+    for i, c in reversed(list(enumerate(chars))):  # reverse so we store the lowest for self-complementary codes
+        basemap[ord(c)] = i
+        basemap[ord(c.lower())] = i
+    return chars, basemap
+
+#this will just pull all the sequences from a fasta file
+def get_sequences(path):
+    #return [seq.values.view(np.uint8) for seq in skbio.io.read(path, format='fasta')]
```

Comment:
remove

Reply:
added a .gitignore to handle that -- let me know if it should be done otherwise